hoomd/md/GeneralEnvelope.h

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2009-2022 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#ifndef __GENERAL_ENVELOPE_H__
+#define __GENERAL_ENVELOPE_H__
+
+#ifndef __HIPCC__
+#include <string>
+#endif
+#include <string.h>
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/VectorMath.h"
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+// DEVICE is __host__ __device__ when included in nvcc and blank when included into the host
+// compiler
+#ifdef __HIPCC__
+#define DEVICE __device__
+#define HOSTDEVICE __host__ __device__
+#else
+#define DEVICE
+#define HOSTDEVICE
+#endif
+
+namespace hoomd
+    {
+namespace md
+    {
+
+/*
+  The GeneralEnvelope creates the pair potential modulator.
+*/
```

Comment:
Move this down to right above the `GeneralEnveolope` class. Make it a [doxygen style comment](https://hoomd-blue.readthedocs.io/en/v4.8.2/style.html#id3). Expand the description to explain the general functional form of an envelope and explain how this class interoperates with the other classes.